src/commons/githubAssessments/GitHubMissionDataUtils.ts

```diff
@@ -0,0 +1,180 @@
+      path: questionFolderName
+    });
+
+    if (!Array.isArray(folderContents.data)) {
+      return questions;
+    }
+
+    const hasSavedCode = (folderContents.data as any[]).find(
+      (file: any) => file.name === 'SavedCode.js'
+    );
+
+    // If the question exists, get the data
+    try {
+      const taskDescription = await getContentAsString(
+        repoOwner,
+        repoName,
+        questionFolderName + '/Problem.md',
+        octokit
+      );
+      const starterCode = await getContentAsString(
+        repoOwner,
+        repoName,
+        questionFolderName + '/StarterCode.js',
+        octokit
+      );
+
+      const savedCode = hasSavedCode
+        ? await getContentAsString(
+            repoOwner,
+            repoName,
+            questionFolderName + '/SavedCode.js',
+            octokit
+          )
+        : starterCode;
+
+      const taskData = new TaskData(taskDescription, starterCode, savedCode);
+
+      questions.push(taskData);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  return questions;
+}
+
+export async function getContentAsString(
+  repoOwner: string,
+  repoName: string,
+  filepath: string,
+  octokit: any
+) {
+  const fileInfo = await octokit.repos.getContent({
+    owner: repoOwner,
+    repo: repoName,
+    path: filepath
+  });
+
+  return Buffer.from((fileInfo.data as any).content, 'base64').toString();
+}
+
+function convertMetadataStringToMissionMetadata(metadataString: string) {
+  const missionMetadata = new MissionMetadata();
+  const stringPropsToExtract = ['coverImage', 'kind', 'number', 'title', 'reading', 'webSummary'];
+  const numPropsToExtract = ['sourceVersion'];
+  const datePropsToExtract = ['dueDate'];
+
+  const retVal = parseMetadataProperties<MissionMetadata>(
+    missionMetadata,
+    stringPropsToExtract,
+    numPropsToExtract,
+    datePropsToExtract,
+    metadataString
+  );
+
+  return retVal;
+}
+
+export function parseMetadataProperties<R>(
+  propertyContainer: R,
+  stringProps: string[],
+  numProps: string[],
+  dateProps: string[],
+  metadataString: string
+) {
+  const lines = metadataString.replace(/\r/g, '').split(/\n/);
+
+  lines.forEach(line => {
+    for (let i = 0; i < stringProps.length; i++) {
+      const propName = stringProps[i];
+
+      if (line.startsWith(propName)) {
+        propertyContainer[propName] = line.substr(propName.length + 1);
+        return;
+      }
+    }
+
+    for (let i = 0; i < numProps.length; i++) {
+      const propName = numProps[i];
+      if (line.startsWith(propName)) {
+        propertyContainer[propName] = parseInt(line.substr(propName.length + 1), 10);
+        return;
+      }
+    }
+
+    for (let i = 0; i < numProps.length; i++) {
+      const propName = dateProps[i];
+
+      if (line.startsWith(propName)) {
+        propertyContainer[propName] = new Date(line.substr(propName.length + 1));
+        return;
+      }
+    }
+  });
+
+  return propertyContainer;
+}
```

Comment:
any error checking for this incase the metafile is incomplete or wrong? what happens to the mission loading then?

also, i think this can be rewritten more succinctly and in n time instead of n^2 time. (can have 1 loop over each line and an array containing each 'expected' field - i think this is ok since metafiles should have a standard format anyway)